OpenRA.Game/FileFormats/WavLoader.cs

```diff
@@ -33,32 +33,33 @@ public class WavLoader : ISoundLoader
 		public enum WaveType { Pcm = 0x1, ImaAdpcm = 0x11 }
 		public static WaveType Type { get; private set; }
 
+		bool wavDetected;
+
 		bool LoadSound(Stream s)
 		{
-			var type = s.ReadASCII(4);
```

Comment:
This reintroduces the original problem:
> Please augment, don't just remove. Removing these checks means that the parser will unconditionally treat the file as a wave, and treat bytes 4..8 as a byte count which is then read. This will then probably crash, as this bogus size can easily be larger than the actual file size.

Sure the crash gets caught, but what if it doesn't crash?  You may still accidentally detect an aud or other audio file as a wav full of bogus chunks that are ignored then plays nothing.

The wav file format dictates that the file starts with `RIFF` followed by `WAVE`, so that is what we check to determine whether it is a wav.